### PR TITLE
Add kafka messages for started, completed and failed 

### PR DESCRIPTION
**Pull request by caismith-ch**
Add service for started, completed and failed messages including unit tests and GenerateDocumentResponse required for the messages.
  
**Resolves**: SFA-578

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/MessageService.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package uk.gov.companieshouse.document.generator.consumer.document.service;
+
+import uk.gov.companieshouse.document.generator.consumer.document.models.GenerateDocumentResponse;
+import uk.gov.companieshouse.document.generator.consumer.document.models.avro.DeserialisedKafkaMessage;
+import uk.gov.companieshouse.kafka.message.Message;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/MessageService.java]: ExecutionException is an unused import

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        byte[] startedData = documentGenerationStateAvroSerializer.serialize(started);
+        return createMessage(startedData, STARTED_PRODUCER_TOPIC);
+    }
+
+    @Override
+    public Message createDocumentGenerationFailed(DeserialisedKafkaMessage deserialisedKafkaMessage,
+                                               GenerateDocumentResponse response) throws IOException {
+
+        DocumentGenerationFailed failed = new DocumentGenerationFailed();
+        failed.setId(deserialisedKafkaMessage != null ? deserialisedKafkaMessage.getId() : "");
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java]: Could you check that `deserialisedKafkaMessage` is not null and only perform the setters if that's the case? That would avoid the need for multiple ternary operators.

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    public Message createDocumentGenerationCompleted(DeserialisedKafkaMessage deserialisedKafkaMessage,
+                                                  GenerateDocumentResponse response) throws IOException {
+
+        DocumentGenerationCompleted completed = new DocumentGenerationCompleted();
+
+        completed.setId(deserialisedKafkaMessage.getId());
+        completed.setRequesterId(deserialisedKafkaMessage.getUserId());
+        completed.setDescription(response.getDescription());
+        completed.setDescriptionIdentifier(response.getDescriptionIdentifier());
+        completed.setLocation(response.getLinks());
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java]: Should this only be set to the self link?

File: pom.xml [hunk 4]
```diff
@@ -115,6 +115,11 @@
             <artifactId>commons-io</artifactId>
             <version>${commons-io.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.0</version>
```
Comment (ChrisJBurns) [inline pom.xml]: put version in properties tags up the top of file

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java (hunk 2 above)
Comment (caismith-ch) [reply src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java]: Wouldn't this create more code as you'd need a second check for when `deserialisedKafkaMessage` is null and set to an empty string?

Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java]: Fair point, disregard this one then

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java (hunk 3 above)
Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java]: Disregard this one too.

Comment (caismith-ch) [reply src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/MessageServiceImpl.java]: Spoke to architect, you are correct. Will update to be a links object with a string Location. 👍

**Comment by caismith-ch** [comment]
Comments addressed. Ready for review.

**Pull request merged**